### PR TITLE
[camera][ios] Migrate the camera view module to Swift

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

To move forward with:
- Migrating more complex modules to the new API
- Improving and testing view callbacks
- Having some modules compatible with Fabric

# How

Basically replaced the existing view manager for camera (`EXCameraManager`) with the `CameraViewModule` that is written in Swift and the new API. The underlying view remained unchanged, so it's still the old `EXCamera` in Objective-C.

Changelog entries are not necessary, it's just following up on things from previous PRs.

# Test Plan

`test-suite` tests are passing (except two tests of digital zoom, but they are also not passing with the old view manager). NCL demo seems to work as well.

### CODE REVIEW

File: packages/expo-camera/ios/EXCamera/CameraViewModule.swift [hunk 1]
```diff
@@ -0,0 +1,337 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import AVFoundation
+import ExpoModulesCore
+
+public final class CameraViewModule: Module {
+  public func definition() -> ModuleDefinition {
```
[the remark this answers is not in the document]
Comment (tsapeta) [reply packages/expo-camera/ios/EXCamera/CameraViewModule.swift:7]: I'm intentionally ignoring this — it's actually hard to follow this rule when the definition function is big. Would be great to just ignore this rule in this function but I don't think there is such option in SwiftLint 😞

**Review by lukmccall — APPROVED** [review]

**Review by Kudo — APPROVED** [review]
